packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -282,27 +330,57 @@ export default function vercelServerless({
 							includeFiles,
 							excludeFiles,
 							maxDuration,
+							isr,
 						});
 						routeDefinitions.push({
 							src: route.pattern.source,
 							dest: func,
 						});
 					}
 				} else {
-					await createFunctionFolder({
-						functionName: NODE_PATH,
-						runtime,
-						entry: new URL(_serverEntry, _buildTempFolder),
-						config: _config,
-						logger,
-						NTF_CACHE,
-						includeFiles,
-						excludeFiles,
-						maxDuration,
-					});
+					if (isr) {
+						await createFunctionFolder({
+							functionName: '_isr',
+							runtime,
+							entry: new URL(_serverEntry, _buildTempFolder),
+							config: _config,
+							logger,
+							NTF_CACHE,
+							includeFiles,
+							excludeFiles,
+							maxDuration,
+							isr,
+						});
+					}
+					if (isr === false || (isr && typeof isr === "object" && isr.exclude)) {
+						await createFunctionFolder({
+							functionName: NODE_PATH,
+							runtime,
+							entry: new URL(_serverEntry, _buildTempFolder),
+							config: _config,
+							logger,
+							NTF_CACHE,
+							includeFiles,
+							excludeFiles,
+							maxDuration,
+							isr: false,
+						});
+					}
+					if (isr && typeof isr === "object" && isr.exclude) {
```

Comment:
We keep doing the same checks, can't we create an internal isr object only once from the options of the user?

Reply:
Cleaned up. [Here is the new control flow](https://github.com/withastro/astro/pull/9714/files#diff-071ccfbdfd7b9bc803219179f88ae0bdb227a399bc4a481b7831c82b801d7a17R332-R344)